hydra-redis/classes_objects.py

```diff
@@ -0,0 +1,118 @@
+            # all the operations for the object is stored in method
+            for obj1 in api_doc.parsed_classes[obj]["class"].supportedProperty:
+                properties.append(obj1.title)
+                if obj1.prop.replace("vocab:", "") in api_doc.parsed_classes:
+                    obj_prop.append(obj1.prop.replace("vocab:", ""))
+            objects_properties["property"] = str(properties)
+            # all property which itself is an object
+            obj_properties_classlist = obj_prop
+            objects_properties["class_property"] = str(
+                obj_properties_classlist)
+            print(objects_node.alias + str(obj))
+            obj_alias = str(objects_node.alias + str(obj)).lower()
+            # key for the node of the object
+            print(obj_alias)
+            obj_node = Node(
+                label="object",
+                alias=obj_alias,
+                properties=objects_properties)
+            self.redis_graph.add_node(obj_node)
+            edge = Edge(objects_node, "has" + str(obj), obj_node)
+            self.redis_graph.add_edge(edge)
+            # set edge between the object and its parent object
+#            print("commit object property")
+#            self.redis_graph.commit()
+            # create the graph in redis
+            if obj_properties_classlist:
+                self.objects_property(
+                    obj_node, obj_properties_classlist, url_node, api_doc)
+
+    def endpointclasses(self, classes_endpoint, url_node, api_doc, base_url):
+        """Node for every class which have an endpoint."""
+        print("classes endpoint or accessing classes")
+
+        for obj in classes_endpoint:
+            obj_method = []
+            obj_prop = []
+            properties = []
+            obj_properties = {}
+            member = {}
+            print("check endpoint url....loading data")
+            new_url = base_url + classes_endpoint[obj]
+            # url for the classes endpoint
+            print(new_url)
+            response = ur.urlopen(new_url)
+            new_file = json.loads(response.read().decode('utf-8'))
+            # retreiving data for the classes endpoint from server
+            if obj in api_doc.parsed_classes:
+                for obj1 in api_doc.parsed_classes[obj]["class"
+                                                        ].supportedOperation:
+                    obj_method.append(obj1.method)
+                print("classes ", obj_method)
+                obj_properties["methods"] = str(obj_method)
+                # store the operations for the endpoint
+                for obj1 in api_doc.parsed_classes[obj]["class"
+                                                        ].supportedProperty:
+                    properties.append(obj1.title)
+                    if obj1.prop.replace(
+                            "vocab:", "") in api_doc.parsed_classes:
+                        obj_prop.append(obj1.prop.replace("vocab:", ""))
```

Comment:
WHY???

First of all, most of the properties won't have `vocab:` in them. They can be schema.org references lilke `'http://schema.org/geo'`. Secondly, this if will always return True.

You should use this to make the graph more connected. For example:
Suppose we have `'vocab:State'` as a prop then you know it'll an instance of `state` class and if State also happens to be an endpoint class then you can just store the URI reference to that object instead of storing the whole object. If State is not an endpoint then you need to parse `state` supportedProperties first in order to store the object.

Reply:
here we are storing those properties which are an object itself And that's type of property has title like `vocab:title`.

Reply:
Yes, I know. That's what I said in the comment above. Please read it carefully.